src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -59,5 +59,10 @@ public function isValid();
      * @param mixed            $i
      * @throws \JsonSchema\Exception\ExceptionInterface
      */
+    public function _check(&$value, $schema = null, JsonPointer $path = null, $i = null);
```

Comment:
> I wrote this PR because I thought that your suggestion was to deprecate coerce, unify behavior with check, and expose checkMode in the public API - and this PR seemed like the best way to go about it - but have I misunderstood your original intention?

Nope, you nailed it. Here's my comment from the other thread:

>>Tell you what--as long as we're going to change the API, let's go all the way and just get rid of coerce, and convert that (back) to a config flag, as well. Then we'll modify check to just take a reference everywhere, and we'll be all set.